### ISSUE TITLE
Missing motivation for TF hacking in Pix2Pix Generative tutorial

### ISSUE THREAD

**Issue by Antymon** [tensorflow/tensorflow#43929]
## URL(s) with the issue:

Please provide a link to the documentation entry, for example:
The note from
https://www.tensorflow.org/tutorials/generative/pix2pix#generate_images

## Description of issue (what needs changing):

Under the section pointed to above there is a note addressing a hack in the code where training mode is used for a piece of code serving as "visual validation". To me, as a person learning from this tutorial, it is insufficient. Although it mitigates the uncertainty around unexpected measures by saying that this is done for the sake of certain shapes of statistics, it also cuts abruptly by saying "and this is what we [don't] want". This leaves me puzzled with why do we want such thing, the more so that further on we take just a single sample which means that the batch norm degenerates to instance norm. 

### Clear description

Can someone please add motivation to why training statistics are preferred in this case and perhaps relate to the issue of BN in the case of a single example?

**Comment by yashk2810** [comment]
See section 3.3 in the paper: https://arxiv.org/pdf/1611.07004.pdf

**Comment by rmothukuru** [comment]
@Antymon,
Can you please respond to the above comment? Thanks!

**Comment by Antymon** [comment]
> @Antymon,
> Can you please respond to the above comment? Thanks!

Sure! Thank you for the reference @yashk2810, that does shed some light indeed!

@rmothukuru As for the issue - I haven't noticed any change in the tutorial contents, or have I missed something? If you want to help just me that's fine, but if you want to enlighten more people perhaps you would like to revisit the contents I have referred to above. 

Thank you!

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
